Services/Membership/classes/class.ilParticipants.php

```diff
@@ -201,6 +201,51 @@ public static function hasParticipantListAccess($a_obj_id, $a_usr_id = null)
         return false;
     }
 
+    /**
+     * This method was introduced as a band-aid fix for #22764.
+     * Please do not use this anywhere else.
+     */
+    public static function canSendMailToMembers(
+        int $ref_id,
+        ?int $usr_id = null,
+        ?int $mail_obj_ref_id = null
+    ) : bool {
+        global $DIC;
+
+        $access = $DIC->access();
+        $rbacsystem = $DIC->rbac()->system();
+
+        if (is_null($usr_id)) {
+            $usr_id = $DIC->user()->getId();
+        }
+        if (is_null($mail_obj_ref_id)) {
+            $mail_obj_ref_id = (new ilMail($usr_id))->getMailObjectReferenceId();
+        }
+
+        if (
+            $access->checkAccess('manage_members', '', $ref_id) &&
+            $rbacsystem->checkAccess('internal_mail', $mail_obj_ref_id)
+        ) {
+            return true;
+        }
+
+        $part = self::getInstance($ref_id);
+        if (!$part->isAssigned($usr_id)) {
+            return false;
+        }
+
+        $object = ilObjectFactory::getInstanceByObjId($obj_id);
```

Comment:
@schmitz-ilias I tried to consume this new method, and found an issue:

* Where does the `$obj_id` come from?
* And: The contact service already has the instance. Creating a new one would cause obsolete SQL queries for N courses/groups. Maybe the instance can also be passed instead of the $ref_id.

A dynamic `$ref_id_or_instance` parameter would be good here.